theano/tensor/elemwise.py

```diff
@@ -1059,35 +1076,110 @@ def _c_all(self, node, nodename, inames, onames, sub):
                                      fortran=alloc_fortran)
             alloc += cgen.make_checks([list(range(nnested))], [odtype],
                                       dict(sub, lv0=oname))
-        olv_index = i  # index of the last output
+        # index of the last output
+        olv_index = i
+
+        # Array referencing all inputs
+        ilength = len(_inames)
+        inArray = """
+        PyArrayObject **inRef[%(ilength)s];
+        """ % locals()
+        for iname, j in izip(_inames, range(len(_inames))):
+            inArray += """
+            inRef[%(j)i]=&%(iname)s;
+            """ % locals()
+
+        # Array referencing all outputs
+        olength = len(_onames)
+        outArray = """
+        PyArrayObject **outRef[%(olength)s];
+        """ % locals()
+        for oname, j in izip(_onames, range(len(onames))):
+            outArray += """
+            outRef[%(j)i]=&%(oname)s;
+            """ % locals()
+
+        alloc += outArray + inArray
```

Comment:
I would do "inArray + outArray" just to keep the same order as they where created.